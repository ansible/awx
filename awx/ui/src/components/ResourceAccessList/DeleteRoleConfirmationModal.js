import React, { Fragment } from 'react';
import { func, string } from 'prop-types';
import { Button } from '@patternfly/react-core';

import { t } from '@lingui/macro';

import { Role } from 'types';
import AlertModal from '../AlertModal';

function DeleteRoleConfirmationModal({ role, username, onCancel, onConfirm }) {
  const isTeamRole = () =>
    typeof role.team_id !== 'undefined' ? t`Team` : t`User`;

  const title = t`Remove ${isTeamRole()} Access`;
  return (
    <AlertModal
      variant="danger"
      title={title}
      isOpen
      onClose={onCancel}
      actions={[
        <Button
          ouiaId="delete-role-modal-delete-button"
          key="delete"
          variant="danger"
          aria-label={t`Confirm delete`}
          onClick={onConfirm}
        >
          {t`Delete`}
        </Button>,
        <Button
          ouiaId="delete-role-modal-cancel-button"
          key="cancel"
          variant="link"
          onClick={onCancel}
        >
          {t`Cancel`}
        </Button>,
      ]}
    >
      {isTeamRole() ? (
        <>
          {t`Are you sure you want to remove ${role.name} access from ${role.team_name}?  Doing so affects all members of the team.`}
          <br />
          <br />
          {t`If you only want to remove access for this particular user, please remove them from the team.`}
        </>
      ) : (
        <>
          {t`Are you sure you want to remove ${role.name} access from ${username}?`}
        </>
      )}
    </AlertModal>
  );
}

DeleteRoleConfirmationModal.propTypes = {
  role: Role.isRequired,
  username: string,
  onCancel: func.isRequired,
  onConfirm: func.isRequired,
};

DeleteRoleConfirmationModal.defaultProps = {
  username: '',
};

export default DeleteRoleConfirmationModal;
