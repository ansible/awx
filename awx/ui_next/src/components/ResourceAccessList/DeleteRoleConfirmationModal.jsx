import React, { Fragment } from 'react';
import { func, string } from 'prop-types';
import { Button } from '@patternfly/react-core';
import { withI18n } from '@lingui/react';
import { t } from '@lingui/macro';

import AlertModal from '../AlertModal';
import { Role } from '../../types';

class DeleteRoleConfirmationModal extends React.Component {
  static propTypes = {
    role: Role.isRequired,
    username: string,
    onCancel: func.isRequired,
    onConfirm: func.isRequired,
  };

  static defaultProps = {
    username: '',
  };

  isTeamRole() {
    const { role } = this.props;
    return typeof role.team_id !== 'undefined';
  }

  render() {
    const { role, username, onCancel, onConfirm, i18n } = this.props;
    const title = i18n._(
      t`Remove ${this.isTeamRole() ? i18n._(t`Team`) : i18n._(t`User`)} Access`
    );
    return (
      <AlertModal
        variant="danger"
        title={title}
        isOpen
        onClose={onCancel}
        actions={[
          <Button
            key="delete"
            variant="danger"
            aria-label={i18n._(t`Confirm delete`)}
            onClick={onConfirm}
          >
            {i18n._(t`Delete`)}
          </Button>,
          <Button key="cancel" variant="secondary" onClick={onCancel}>
            {i18n._(t`Cancel`)}
          </Button>,
        ]}
      >
        {this.isTeamRole() ? (
          <Fragment>
            {i18n._(
              t`Are you sure you want to remove ${role.name} access from ${role.team_name}?  Doing so affects all members of the team.`
            )}
            <br />
            <br />
            {i18n._(
              t`If you only want to remove access for this particular user, please remove them from the team.`
            )}
          </Fragment>
        ) : (
          <Fragment>
            {i18n._(
              t`Are you sure you want to remove ${role.name} access from ${username}?`
            )}
          </Fragment>
        )}
      </AlertModal>
    );
  }
}

export default withI18n()(DeleteRoleConfirmationModal);
