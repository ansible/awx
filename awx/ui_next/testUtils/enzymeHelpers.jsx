/*
 * Enzyme helpers for injecting top-level contexts
 * derived from https://lingui.js.org/guides/testing.html
 */
import React from 'react';
import { shape, object, string, arrayOf } from 'prop-types';
import { mount, shallow } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';
import { I18nProvider } from '@lingui/react';
import { ConfigProvider } from '../src/contexts/Config';

const language = 'en-US';
const intlProvider = new I18nProvider(
  {
    language,
    catalogs: {
      [language]: {},
    },
  },
  {}
);
const {
  linguiPublisher: { i18n: originalI18n },
} = intlProvider.getChildContext();

const defaultContexts = {
  linguiPublisher: {
    i18n: {
      ...originalI18n,
      _: key => key.id, // provide _ macro, for just passing down the key
      toJSON: () => '/i18n/',
    },
  },
  config: {
    ansible_version: null,
    custom_virtualenvs: [],
    version: null,
    toJSON: () => '/config/',
  },
  router: {
    history_: {
      push: () => {},
      replace: () => {},
      createHref: () => {},
      listen: () => {},
      location: {
        hash: '',
        pathname: '',
        search: '',
        state: '',
      },
      toJSON: () => '/history/',
    },
    route: {
      location: {
        hash: '',
        pathname: '',
        search: '',
        state: '',
      },
      match: {
        params: {},
        isExact: false,
        path: '',
        url: '',
      },
    },
    toJSON: () => '/router/',
  },
};

function wrapContexts(node, context) {
  const { config, router } = context;
  class Wrap extends React.Component {
    render() {
      // eslint-disable-next-line react/no-this-in-sfc
      const { children, ...props } = this.props;
      const component = React.cloneElement(children, props);
      if (router.history) {
        return (
          <ConfigProvider value={config}>
            <Router history={router.history}>{component}</Router>
          </ConfigProvider>
        );
      }
      return (
        <ConfigProvider value={config}>
          <MemoryRouter>{component}</MemoryRouter>
        </ConfigProvider>
      );
    }
  }

  return <Wrap>{node}</Wrap>;
}

function applyDefaultContexts(context) {
  if (!context) {
    return defaultContexts;
  }
  const newContext = {};
  Object.keys(defaultContexts).forEach(key => {
    newContext[key] = {
      ...defaultContexts[key],
      ...context[key],
    };
  });
  return newContext;
}

export function shallowWithContexts(node, options = {}) {
  const context = applyDefaultContexts(options.context);
  return shallow(wrapContexts(node, context));
}

export function mountWithContexts(node, options = {}) {
  const context = applyDefaultContexts(options.context);
  const childContextTypes = {
    linguiPublisher: shape({
      i18n: object.isRequired,
    }).isRequired,
    config: shape({
      ansible_version: string,
      custom_virtualenvs: arrayOf(string),
      version: string,
    }),
    router: shape({
      route: shape({
        location: shape({}),
        match: shape({}),
      }).isRequired,
      history: shape({}),
    }),
    ...options.childContextTypes,
  };
  return mount(wrapContexts(node, context), { context, childContextTypes });
}

/**
 * Wait for element(s) to achieve a desired state.
 *
 * @param[wrapper] - A ReactWrapper instance
 * @param[selector] - The selector of the element(s) to wait for.
 * @param[callback] - Callback to poll - by default this checks for a node count of 1.
 */
export function waitForElement(
  wrapper,
  selector,
  callback = el => el.length === 1
) {
  const interval = 100;
  return new Promise((resolve, reject) => {
    let attempts = 30;
    (function pollElement() {
      wrapper.update();
      const el = wrapper.find(selector);
      if (callback(el)) {
        return resolve(el);
      }
      if (--attempts <= 0) {
        const message = `Expected condition for <${selector}> not met: ${callback.toString()}`;
        return reject(new Error(message));
      }
      return setTimeout(pollElement, interval);
    })();
  });
}
