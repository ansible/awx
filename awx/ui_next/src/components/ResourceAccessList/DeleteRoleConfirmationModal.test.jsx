import React from 'react';

import { mountWithContexts } from '@testUtils/enzymeHelpers';

import DeleteRoleConfirmationModal from './DeleteRoleConfirmationModal';

const role = {
  id: 3,
  name: 'Member',
  resource_name: 'Org',
  resource_type: 'organization',
  team_id: 5,
  team_name: 'The Team',
};

describe('<DeleteRoleConfirmationModal />', () => {
  test('should render initially', () => {
    const wrapper = mountWithContexts(
      <DeleteRoleConfirmationModal
        role={role}
        username="jane"
        onCancel={() => {}}
        onConfirm={() => {}}
      />
    );
    wrapper.update();
    expect(wrapper.find('DeleteRoleConfirmationModal')).toMatchSnapshot();
  });
});
