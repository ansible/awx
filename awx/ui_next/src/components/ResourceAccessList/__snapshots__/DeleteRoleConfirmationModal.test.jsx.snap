// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteRoleConfirmationModal /> should render initially 1`] = `
<DeleteRoleConfirmationModal
  i18n={"/i18n/"}
  onCancel={[Function]}
  onConfirm={[Function]}
  role={
    Object {
      "id": 3,
      "name": "Member",
      "resource_name": "Org",
      "resource_type": "organization",
      "team_id": 5,
      "team_name": "The Team",
    }
  }
  username="jane"
>
  <AlertModal
    actions={
      Array [
        <Unknown
          aria-label="Confirm delete"
          onClick={[Function]}
          variant="danger"
        >
          Delete
        </Unknown>,
        <Unknown
          onClick={[Function]}
          variant="secondary"
        >
          Cancel
        </Unknown>,
      ]
    }
    isOpen={true}
    onClose={[Function]}
    title="Remove Team Access"
    variant="danger"
  >
    <Modal
      actions={
        Array [
          <Unknown
            aria-label="Confirm delete"
            onClick={[Function]}
            variant="danger"
          >
            Delete
          </Unknown>,
          <Unknown
            onClick={[Function]}
            variant="secondary"
          >
            Cancel
          </Unknown>,
        ]
      }
      appendTo={
        <body
          class="pf-c-backdrop__open"
        >
          <div>
            <div
              class="pf-c-backdrop"
            >
              <div
                class="pf-l-bullseye"
              >
                <div
                  aria-describedby="pf-modal-0"
                  aria-label="Remove Team Access"
                  aria-modal="true"
                  class="pf-c-modal-box pf-m-sm"
                  role="dialog"
                >
                  <button
                    aria-label="Close"
                    class="pf-c-button pf-m-plain"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                        transform=""
                      />
                    </svg>
                  </button>
                  <div
                    class="pf-c-title"
                  >
                    <div
                      class="AlertModal__Header-sc-9waqvl-0 dYqVFx"
                    >
                      <svg
                        aria-hidden="true"
                        class="sc-bdVaJa hIExGP"
                        fill="currentColor"
                        height="2em"
                        role="img"
                        style="vertical-align: -0.25em;"
                        viewBox="0 0 512 512"
                        width="2em"
                      >
                        <path
                          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                          transform=""
                        />
                      </svg>
                      <h1
                        class="pf-c-title pf-m-2xl"
                      >
                        Remove Team Access
                      </h1>
                    </div>
                  </div>
                  <div
                    class="pf-c-modal-box__body"
                    id="pf-modal-0"
                  >
                    Are you sure you want to remove Member access from The Team?  Doing so affects all members of the team.
                    <br />
                    <br />
                    If you only want to remove access for this particular user, please remove them from the team.
                  </div>
                  <div
                    class="pf-c-modal-box__footer pf-m-align-left"
                  >
                    <button
                      aria-label="Confirm delete"
                      class="pf-c-button pf-m-danger"
                      type="button"
                    >
                      Delete
                    </button>
                    <button
                      class="pf-c-button pf-m-secondary"
                      type="button"
                    >
                      Cancel
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </body>
      }
      ariaDescribedById=""
      className=""
      header={
        <AlertModal__Header>
          <Styled(ExclamationCircleIcon)
            size="lg"
          />
          <Title
            size="2xl"
          >
            Remove Team Access
          </Title>
        </AlertModal__Header>
      }
      hideTitle={false}
      isFooterLeftAligned={true}
      isLarge={false}
      isOpen={true}
      isSmall={true}
      onClose={[Function]}
      showClose={true}
      title="Remove Team Access"
    >
      <Portal
        containerInfo={
          <div>
            <div
              class="pf-c-backdrop"
            >
              <div
                class="pf-l-bullseye"
              >
                <div
                  aria-describedby="pf-modal-0"
                  aria-label="Remove Team Access"
                  aria-modal="true"
                  class="pf-c-modal-box pf-m-sm"
                  role="dialog"
                >
                  <button
                    aria-label="Close"
                    class="pf-c-button pf-m-plain"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                        transform=""
                      />
                    </svg>
                  </button>
                  <div
                    class="pf-c-title"
                  >
                    <div
                      class="AlertModal__Header-sc-9waqvl-0 dYqVFx"
                    >
                      <svg
                        aria-hidden="true"
                        class="sc-bdVaJa hIExGP"
                        fill="currentColor"
                        height="2em"
                        role="img"
                        style="vertical-align: -0.25em;"
                        viewBox="0 0 512 512"
                        width="2em"
                      >
                        <path
                          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                          transform=""
                        />
                      </svg>
                      <h1
                        class="pf-c-title pf-m-2xl"
                      >
                        Remove Team Access
                      </h1>
                    </div>
                  </div>
                  <div
                    class="pf-c-modal-box__body"
                    id="pf-modal-0"
                  >
                    Are you sure you want to remove Member access from The Team?  Doing so affects all members of the team.
                    <br />
                    <br />
                    If you only want to remove access for this particular user, please remove them from the team.
                  </div>
                  <div
                    class="pf-c-modal-box__footer pf-m-align-left"
                  >
                    <button
                      aria-label="Confirm delete"
                      class="pf-c-button pf-m-danger"
                      type="button"
                    >
                      Delete
                    </button>
                    <button
                      class="pf-c-button pf-m-secondary"
                      type="button"
                    >
                      Cancel
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <ModalContent
          actions={
            Array [
              <Unknown
                aria-label="Confirm delete"
                onClick={[Function]}
                variant="danger"
              >
                Delete
              </Unknown>,
              <Unknown
                onClick={[Function]}
                variant="secondary"
              >
                Cancel
              </Unknown>,
            ]
          }
          ariaDescribedById=""
          className=""
          header={
            <AlertModal__Header>
              <Styled(ExclamationCircleIcon)
                size="lg"
              />
              <Title
                size="2xl"
              >
                Remove Team Access
              </Title>
            </AlertModal__Header>
          }
          hideTitle={false}
          id="pf-modal-0"
          isFooterLeftAligned={true}
          isLarge={false}
          isOpen={true}
          isSmall={true}
          onClose={[Function]}
          showClose={true}
          title="Remove Team Access"
        >
          <Backdrop>
            <div
              className="pf-c-backdrop"
            >
              <FocusTrap
                active={true}
                className="pf-l-bullseye"
                focusTrapOptions={
                  Object {
                    "clickOutsideDeactivates": true,
                  }
                }
                paused={false}
              >
                <div
                  className="pf-l-bullseye"
                >
                  <ModalBox
                    className=""
                    id="pf-modal-0"
                    isLarge={false}
                    isSmall={true}
                    style={Object {}}
                    title="Remove Team Access"
                  >
                    <div
                      aria-describedby="pf-modal-0"
                      aria-label="Remove Team Access"
                      aria-modal="true"
                      className="pf-c-modal-box pf-m-sm"
                      role="dialog"
                      style={Object {}}
                    >
                      <ModalBoxCloseButton
                        onClose={[Function]}
                      >
                        <Component
                          aria-label="Close"
                          className=""
                          onClick={[Function]}
                          variant="plain"
                        >
                          <ComponentWithOuia
                            component={[Function]}
                            componentProps={
                              Object {
                                "aria-label": "Close",
                                "children": <TimesIcon
                                  color="currentColor"
                                  noVerticalAlign={false}
                                  size="sm"
                                  title={null}
                                />,
                                "className": "",
                                "onClick": [Function],
                                "variant": "plain",
                              }
                            }
                            consumerContext={null}
                          >
                            <Button
                              aria-label="Close"
                              className=""
                              onClick={[Function]}
                              ouiaContext={
                                Object {
                                  "isOuia": false,
                                  "ouiaId": null,
                                }
                              }
                              variant="plain"
                            >
                              <button
                                aria-disabled={null}
                                aria-label="Close"
                                className="pf-c-button pf-m-plain"
                                disabled={false}
                                onClick={[Function]}
                                tabIndex={null}
                                type="button"
                              >
                                <TimesIcon
                                  color="currentColor"
                                  noVerticalAlign={false}
                                  size="sm"
                                  title={null}
                                >
                                  <svg
                                    aria-hidden={true}
                                    aria-labelledby={null}
                                    fill="currentColor"
                                    height="1em"
                                    role="img"
                                    style={
                                      Object {
                                        "verticalAlign": "-0.125em",
                                      }
                                    }
                                    viewBox="0 0 352 512"
                                    width="1em"
                                  >
                                    <path
                                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                      transform=""
                                    />
                                  </svg>
                                </TimesIcon>
                              </button>
                            </Button>
                          </ComponentWithOuia>
                        </Component>
                      </ModalBoxCloseButton>
                      <div
                        className="pf-c-title"
                      >
                        <AlertModal__Header>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "AlertModal__Header-sc-9waqvl-0",
                                  "isStatic": true,
                                  "lastClassName": "dYqVFx",
                                  "rules": Array [
                                    "display:flex;svg{margin-right:16px;}",
                                  ],
                                },
                                "displayName": "AlertModal__Header",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "AlertModal__Header-sc-9waqvl-0",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="AlertModal__Header-sc-9waqvl-0 dYqVFx"
                            >
                              <Styled(ExclamationCircleIcon)
                                size="lg"
                              >
                                <StyledComponent
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-bdVaJa",
                                        "isStatic": true,
                                        "lastClassName": "hIExGP",
                                        "rules": Array [
                                          "color: var(--pf-global--danger-color--100)",
                                        ],
                                      },
                                      "displayName": "Styled(ExclamationCircleIcon)",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "sc-bdVaJa",
                                      "target": [Function],
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  size="lg"
                                >
                                  <ExclamationCircleIcon
                                    className="sc-bdVaJa hIExGP"
                                    color="currentColor"
                                    noVerticalAlign={false}
                                    size="lg"
                                    title={null}
                                  >
                                    <svg
                                      aria-hidden={true}
                                      aria-labelledby={null}
                                      className="sc-bdVaJa hIExGP"
                                      fill="currentColor"
                                      height="2em"
                                      role="img"
                                      style={
                                        Object {
                                          "verticalAlign": "-0.25em",
                                        }
                                      }
                                      viewBox="0 0 512 512"
                                      width="2em"
                                    >
                                      <path
                                        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                                        transform=""
                                      />
                                    </svg>
                                  </ExclamationCircleIcon>
                                </StyledComponent>
                              </Styled(ExclamationCircleIcon)>
                              <Title
                                size="2xl"
                              >
                                <h1
                                  className="pf-c-title pf-m-2xl"
                                >
                                  Remove Team Access
                                </h1>
                              </Title>
                            </div>
                          </StyledComponent>
                        </AlertModal__Header>
                      </div>
                      <ModalBoxBody
                        id="pf-modal-0"
                      >
                        <div
                          className="pf-c-modal-box__body"
                          id="pf-modal-0"
                        >
                          Are you sure you want to remove Member access from The Team?  Doing so affects all members of the team.
                          <br />
                          <br />
                          If you only want to remove access for this particular user, please remove them from the team.
                        </div>
                      </ModalBoxBody>
                      <ModalBoxFooter
                        isLeftAligned={true}
                      >
                        <div
                          className="pf-c-modal-box__footer pf-m-align-left"
                        >
                          <Component
                            aria-label="Confirm delete"
                            key="delete"
                            onClick={[Function]}
                            variant="danger"
                          >
                            <ComponentWithOuia
                              component={[Function]}
                              componentProps={
                                Object {
                                  "aria-label": "Confirm delete",
                                  "children": "Delete",
                                  "onClick": [Function],
                                  "variant": "danger",
                                }
                              }
                              consumerContext={null}
                            >
                              <Button
                                aria-label="Confirm delete"
                                onClick={[Function]}
                                ouiaContext={
                                  Object {
                                    "isOuia": false,
                                    "ouiaId": null,
                                  }
                                }
                                variant="danger"
                              >
                                <button
                                  aria-disabled={null}
                                  aria-label="Confirm delete"
                                  className="pf-c-button pf-m-danger"
                                  disabled={false}
                                  onClick={[Function]}
                                  tabIndex={null}
                                  type="button"
                                >
                                  Delete
                                </button>
                              </Button>
                            </ComponentWithOuia>
                          </Component>
                          <Component
                            key="cancel"
                            onClick={[Function]}
                            variant="secondary"
                          >
                            <ComponentWithOuia
                              component={[Function]}
                              componentProps={
                                Object {
                                  "children": "Cancel",
                                  "onClick": [Function],
                                  "variant": "secondary",
                                }
                              }
                              consumerContext={null}
                            >
                              <Button
                                onClick={[Function]}
                                ouiaContext={
                                  Object {
                                    "isOuia": false,
                                    "ouiaId": null,
                                  }
                                }
                                variant="secondary"
                              >
                                <button
                                  aria-disabled={null}
                                  aria-label={null}
                                  className="pf-c-button pf-m-secondary"
                                  disabled={false}
                                  onClick={[Function]}
                                  tabIndex={null}
                                  type="button"
                                >
                                  Cancel
                                </button>
                              </Button>
                            </ComponentWithOuia>
                          </Component>
                        </div>
                      </ModalBoxFooter>
                    </div>
                  </ModalBox>
                </div>
              </FocusTrap>
            </div>
          </Backdrop>
        </ModalContent>
      </Portal>
    </Modal>
  </AlertModal>
</DeleteRoleConfirmationModal>
`;
